include/RAJA/policy/cuda/kernel/CudaKernel.hpp

```diff
@@ -314,6 +319,23 @@ struct CudaLaunchHelper<cuda_launch<async0, num_blocks, num_threads>,StmtList,Da
       internal::cuda_occupancy_max_blocks<Self>(
           func, shmem_size, max_blocks, actual_threads);
 
+      if(max_blocks == 0) {
+
+        //
+        // actual_threads is too high for the kernel to run
+        // successfully due to resource constraints reset to
+        // something that is possible
+        //
+        int min_grid_size, max_block_size;
+        // NOTE: min_grid_size is the minimum grid size needed
+        //       to achieve the best potential occupancy
+        cudaErrchk(cudaOccupancyMaxPotentialBlockSize(
```

Comment:
The cudaOccupancy functions add a lot of overhead so we cache the results in our own functions.
Please change this to use our function instead of the cudaOccupancy function directly.
https://github.com/LLNL/RAJA/blob/6b0e35d821059ef33f8bb6442b18a4e6f58d4231/include/RAJA/policy/cuda/kernel/internal.hpp#L221

Reply:
Ok, I added this change.  I didn't realize that there was an internal function that did this and cached the result.